proxy/agent.go

```diff
@@ -78,6 +78,12 @@ var (
 	}
 )
 
+const (
+	// MaxAborts is the maximum number of cascading abort messages to buffer.
+	// This should be the upper bound on the number of proxies available at any point in time.
+	MaxAborts = 10
```

Comment:
that's an undocumented behavior of envoy that it non-gracefully kills (n-2)-nd epoch.